core/src/main/java/io/grpc/internal/DelayedClientTransport2.java

```diff
@@ -105,54 +144,63 @@ public final Runnable start(Listener listener) {
   @Override
   public final ClientStream newStream(MethodDescriptor<?, ?> method, Metadata headers,
       CallOptions callOptions, StatsTraceContext statsTraceCtx) {
-    SubchannelPicker picker = null;
-    synchronized (lock) {
-      if (!shutdown) {
-        if (lastPicker == null) {
-          return createPendingStream(method, headers, callOptions, statsTraceCtx);
+    try {
+      SubchannelPicker picker = null;
+      long pickerVersion = -1;
+      synchronized (lock) {
+        if (!shutdown) {
+          if (lastPicker == null) {
+            return createPendingStream(method, headers, callOptions, statsTraceCtx);
+          }
+          picker = lastPicker;
+          pickerVersion = lastPickerVersion;
         }
-        picker = lastPicker;
       }
-    }
-    if (picker != null) {
-      while (true) {
-        PickResult pickResult = picker.pickSubchannel(callOptions.getAffinity(), headers);
-        ClientTransport transport = GrpcUtil.getTransportFromPickResult(
-            pickResult, callOptions.isWaitForReady());
-        if (transport != null) {
-          return transport.newStream(method, headers, callOptions, statsTraceCtx);
-        }
-        // This picker's conclusion is "buffer".  If there hasn't been a newer picker set
-        // (possible race with reprocess()), we will buffer it.  Otherwise, will try with the new
-        // picker.
-        synchronized (lock) {
-          if (shutdown) {
-            break;
+      if (picker != null) {
+        while (true) {
+          PickResult pickResult = picker.pickSubchannel(callOptions.getAffinity(), headers);
+          ClientTransport transport = GrpcUtil.getTransportFromPickResult(
+              pickResult, callOptions.isWaitForReady());
+          if (transport != null) {
+            return transport.newStream(method, headers, callOptions, statsTraceCtx);
```

Comment:
This isn't new in this PR, but I think this may be a bug: if the transport returns `null` here, it seems we should buffer. (No need to change in this PR)